rlclientlib/live_model_impl.cc

```diff
@@ -417,4 +423,35 @@ namespace reinforcement_learning {
     return error_code::success;
   }
 
+  int post_process_rank(ranking_response& response, decision_mode decision_mode) {
+    if (decision_mode == IMITATION_MODE) {
+      std::vector<action_prob> copied_action_prob;
```

Comment:
1) This will allocate memory for every call.  
2) it also extra copies of memory
response already has a vector.  please reuse this instead.

Reply:
The problem there is that response's vector has a constant iterator - I cannot shift the order unless I clear them.

Checking to remove the constant iterator but need to confirm the reason when adding it



---
In reply to: [385784544](https://github.com/VowpalWabbit/reinforcement_learning/pull/129#discussion_r385784544) [](ancestors = 385784544)

Reply:
Now it has changed to in-memory replacement. Please see the change in `container_iterator.h`